dll/cpl/desk/settings.c

```diff
@@ -866,10 +898,14 @@ ApplyDisplaySettings(HWND hwndDlg, PSETTINGS_DATA pData)
     }
     else
     {
-        pData->CurrentDisplayDevice->CurrentSettings->dmPelsWidth = pData->CurrentDisplayDevice->InitialSettings.dmPelsWidth;
-        pData->CurrentDisplayDevice->CurrentSettings->dmPelsHeight = pData->CurrentDisplayDevice->InitialSettings.dmPelsHeight;
-        pData->CurrentDisplayDevice->CurrentSettings->dmBitsPerPel = pData->CurrentDisplayDevice->InitialSettings.dmBitsPerPel;
-        pData->CurrentDisplayDevice->CurrentSettings->dmDisplayFrequency = pData->CurrentDisplayDevice->InitialSettings.dmDisplayFrequency;
+        PSETTINGS_ENTRY pInitialSettings = &pData->CurrentDisplayDevice->InitialSettings;
+        pData->CurrentDisplayDevice->CurrentSettings =
+            FindBestElement(pInitialSettings->dmPelsWidth,
+                            pInitialSettings->dmPelsHeight,
+                            pInitialSettings->dmBitsPerPel,
+                            pData->CurrentDisplayDevice->CurrentSettings);
+        /* Initial entry should be found */
+        ASSERT(pData->CurrentDisplayDevice->CurrentSettings);
```

Comment:
Unless some other usages are planned, move `ASSERT()` to the function.

Reply:
Moved.